tests/test_pulse.py

```diff
@@ -0,0 +1,70 @@
+import numpy as np
+import pytest
+from inline_snapshot import snapshot
+
+from promdens.promdens import ENVELOPE_TYPES, LaserPulse
+
+def test_invalid_envelope_type(make_pulse):
+    with pytest.raises(ValueError):
+        make_pulse(envelope_type='invalid')
+
+
+@pytest.mark.parametrize("envelope", ENVELOPE_TYPES)
+def test_envelope_types(make_pulse, envelope):
+    # For now just test that we can create a pulse
+    # for all available envelope types, more to come.
+    pulse = make_pulse(envelope_type=envelope)
+
+    assert envelope in str(pulse)
+
+
+def test_field_cos(make_pulse):
+    pulse = make_pulse(omega=0.1, lchirp=0.0)
+    t = np.array([-10., -1.0, 0.0, 1.0, 10.0])
+    cos = pulse.field_cos(t)
+
+    s = snapshot(
```

Comment:
Quoting from [docs](https://15r10nk.github.io/inline-snapshot/#welcome-to-inline-snapshot):

> golden master/approval/snapshot testing. The idea is that you have a function with a currently unknown result and you want to write a tests, which ensures that the result does not change during refactoring.

In our case, we're testing a complicated numerical functions, and this library records the known-good values.